drivers/ethernet/phy/phy_dm8806.c

```diff
@@ -0,0 +1,466 @@
+		return ret;
+	}
+	k_busy_wait(500);
+
+	/* Change the link speed. */
+	ret = mdio_read(cfg->mdio, cfg->phy_addr, PORTX_PHY_CONTROL_REGISTER, &data);
+	if (ret) {
+		LOG_ERR("Failed to read data from DM8806");
+		return ret;
+	}
+	k_busy_wait(500);
+	data &= ~(LINK_SPEED | DUPLEX_MODE);
+	data |= req_speed;
+	ret = mdio_write(cfg->mdio, cfg->phy_addr, PORTX_PHY_CONTROL_REGISTER, data);
+	if (ret) {
+		LOG_ERR("Failed to write data to DM8806");
+		return ret;
+	}
+	k_busy_wait(500);
+
+	/* Power up ethernet port*/
+	ret = mdio_read(cfg->mdio, cfg->phy_addr, PORTX_PHY_CONTROL_REGISTER, &data);
+	if (ret) {
+		LOG_ERR("Failes to read data drom DM8806");
+		return ret;
+	}
+	k_busy_wait(500);
+	data &= ~(POWER_DOWN);
+	ret = mdio_write(cfg->mdio, cfg->phy_addr, PORTX_PHY_CONTROL_REGISTER, data);
+	if (ret) {
+		LOG_ERR("Failed to write data to DM8806");
+		return ret;
+	}
+	k_busy_wait(500);
+	return -ENOTSUP;
+}
+
+static int phy_dm8806_reg_read(const struct device *dev, uint16_t reg_addr, uint32_t *data)
+{
+	int res;
+	const struct phy_dm8806_config *cfg = dev->config;
+
+	res = mdio_read(cfg->mdio, cfg->switch_addr, reg_addr, (uint16_t *)data);
+	if (res) {
+		LOG_ERR("Failed to read data from DM8806");
+		return res;
+	}
+	return res;
+}
+
+static int phy_dm8806_reg_write(const struct device *dev, uint16_t reg_addr, uint32_t data)
+{
+	int res;
+	const struct phy_dm8806_config *cfg = dev->config;
+
+	res = mdio_write(cfg->mdio, cfg->switch_addr, reg_addr, data);
+	if (res) {
+		LOG_ERR("Failed to write data to DM8806");
+		return res;
+	}
+	return res;
+}
+
+static int phy_dm8806_link_cb_set(const struct device *dev, phy_callback_t cb, void *user_data)
+{
+	int res = 0;
+	struct phy_dm8806_data *data = dev->data;
+	const struct phy_dm8806_config *cfg = dev->config;
+
+	res = gpio_pin_interrupt_configure_dt(&cfg->gpio_int, GPIO_INT_DISABLE);
+	if (res != 0) {
+		LOG_WRN("Failed to disable DM8806 interrupt: %i", res);
+		return res;
+	}
+	data->link_speed_chenge_cb = cb;
+	data->cb_data = user_data;
+	gpio_pin_interrupt_configure_dt(&cfg->gpio_int, GPIO_INT_EDGE_TO_ACTIVE);
+	if (res != 0) {
+		LOG_WRN("Failed to enable DM8806 interrupt: %i", res);
+		return res;
+	}
+
+	return res;
+}
+
+static DEVICE_API(ethphy, phy_dm8806_api) = {
+	.get_link = phy_dm8806_get_link_state,
+	.cfg_link = phy_dm8806_cfg_link,
+#ifdef CONFIG_PHY_DM8806_TRIGGER
+	.link_cb_set = phy_dm8806_link_cb_set,
+#endif
+	.read = phy_dm8806_reg_read,
+	.write = phy_dm8806_reg_write,
+};
+
+#define DM8806_PHY_DEFINE_CONFIG(n)                                                                \
+	static const struct phy_dm8806_config phy_dm8806_config_##n = {                            \
+		.mdio = DEVICE_DT_GET(DT_INST_BUS(n)),                                             \
+		.phy_addr = DT_INST_REG_ADDR(n),                                                   \
+		.switch_addr = DT_PROP(DT_NODELABEL(dm8806_phy##n), reg_switch),                   \
```

Comment:
this also looks incorrect, it's making an assumption about a nodelabel, I'd imagine this should have been `DT_INST_PROP(n, reg_switch)`

Reply:
ya it's expecting the device node to have a label for some reason ...

```
dm8806_phy0: ethernet-phy@0 {
			reg = <0>;
			reg-switch = <8>
			...
```